core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestPreferWritePartitioning.java

```diff
@@ -71,38 +86,93 @@ public void tearDown()
         tester = null;
     }
 
+    @Test(dataProvider = "preferWritePartitioningDataProvider")
+    public void testPreferWritePartitioning(Session session, double distinctValuesStat, boolean match)
+    {
+        PlanNodeStatsEstimate stats = PlanNodeStatsEstimate.builder()
```

Comment:
generate `PlanNodeStatsEstimate` in `if (distinctValuesStat != NO_STATS) {` branch